graylog2-server/src/main/java/org/graylog2/indexer/ranges/EsIndexRangeService.java

```diff
@@ -0,0 +1,324 @@
+    private final Indices indices;
+
+    @Inject
+    public EsIndexRangeService(Client client, ActivityWriter activityWriter, ObjectMapper objectMapper, Indices indices) {
+        this.client = client;
+        this.activityWriter = activityWriter;
+        this.objectMapper = objectMapper;
+        this.indices = indices;
+    }
+
+    @Override
+    @Nullable
+    public IndexRange get(String index) throws NotFoundException {
+        final GetRequest request = new GetRequestBuilder(client, index)
+                .setType(IndexMapping.TYPE_META)
+                .setId(index)
+                .request();
+
+        final GetResponse r;
+        try {
+            r = client.get(request).actionGet();
+        } catch (IndexMissingException | NoShardAvailableActionException e) {
+            throw new NotFoundException(e);
+        }
+
+        if (!r.isExists()) {
+            throw new NotFoundException("Index [" + index + "] not found.");
+        }
+
+        return parseSource(r.getIndex(), r.getSource());
+    }
+
+    @Nullable
+    private IndexRange parseSource(String index, Map<String, Object> fields) {
+        try {
+            return IndexRange.create(
+                    index,
+                    parseFromDateString((String) fields.get("begin")),
+                    parseFromDateString((String) fields.get("end")),
+                    parseFromDateString((String) fields.get("calculated_at")),
+                    (int) fields.get("took_ms")
+            );
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    private DateTime parseFromDateString(String s) {
+        return DateTime.parse(s);
+    }
+
+    @Override
+    public SortedSet<IndexRange> find(DateTime begin, DateTime end) {
+        final RangeQueryBuilder beginRangeQuery = QueryBuilders.rangeQuery("begin").gte(begin.getMillis());
+        final RangeQueryBuilder endRangeQuery = QueryBuilders.rangeQuery("end").lte(end.getMillis());
+        final BoolQueryBuilder completeRangeQuery = QueryBuilders.boolQuery()
+                .must(beginRangeQuery)
+                .must(endRangeQuery);
+        final SearchRequest request = client.prepareSearch()
+                .setTypes(IndexMapping.TYPE_META)
+                .setQuery(completeRangeQuery)
+                .request();
+
+        final SearchResponse response = client.search(request).actionGet();
+        final ImmutableSortedSet.Builder<IndexRange> indexRanges = ImmutableSortedSet.orderedBy(IndexRange.COMPARATOR);
+        for (SearchHit searchHit : response.getHits()) {
+            final IndexRange indexRange = parseSource(searchHit.getIndex(), searchHit.getSource());
+            if (indexRange != null) {
+                indexRanges.add(indexRange);
+            }
+        }
+
+        return indexRanges.build();
+    }
+
+    @Override
+    public SortedSet<IndexRange> findAll() {
+        final SearchRequest request = client.prepareSearch()
```

Comment:
I think this search also needs to take into account the graylog index prefix.
Otherwise we could pull in indices that could be part of another graylog deployment writing to the same cluster. (Unlikely but that would trigger very fun bugs :santa: )